Sources/Testing/Testing.docc/testing-for-errors-in-swift-code.md

```diff
@@ -0,0 +1,118 @@
+attributes associated with the item at a given index.  Both of these methods
+throw errors if their parameters are outside the list's range.
+
+```swift
+struct AttributedArray<T> {
+    enum AttributedArrayError : Error {
+        case outOfRange
+    }
+
+    let elements: [T]
+    var attributes: [Int: [String]]
+
+    init(list: [T]) {
+        elements = list
+        attributes = [Int: [String]]()
+    }
+
+    mutating func add(attribute: String, toObjectsIn range: Range<Int>) throws {
+        guard Int(range.startIndex) >= 0 && Int(range.endIndex) < elements.count else {
+            throw AttributedArrayError.outOfRange
+        }
+        for index in range {
+            if var attributeList = attributes[index] {
+                attributeList.append(attribute)
+                attributes[index] = attributeList
+            } else {
+                attributes[index] = [attribute]
+            }
+        }
+    }
+
+    func attributes(forItemAt index: Int) throws -> [String] {
+        guard index >= 0 && index < elements.count else {
+            throw AttributedArrayError.outOfRange
+        }
+        return attributes[index] ?? []
+    }
+
+    // Other methods.
+}
+```
+
+Create a test function that `throws` and `try` the code under test.
+If the code throws an error, then your test fails.
+
+To check that the code under test throws a specific error, or to continue a
+longer test function after the code throws an error, pass that error as the
+first argument of ``expect(throws:_:sourcelocation:performing:)-1xr34``, and
+pass a closure that calls the code under test:
+
+```swift
+@Test func cannotAddAttributeToObjectBeforeStartOfList() {
+    var array = AttributedArray(list: [1,2,3])
+    #expect(throws: AttributedArray<Int>.AttributedArrayError.outOfRange) {
+        try array.add(attribute: "my attribute", toObjectsIn: -1..<0)
+    }
+}
+```
+
+If the closure completes without throwing an error, the testing library
+records an issue.  Other overloads of `#expect()` let you test that
+the code throws an error of a given type, or matching an arbitrary
+Boolean test.  Similar overloads of `#require()` stop running your
+test if the code doesn't throw the expected error.
+
+### Validate that your code doesn't throw an error
+
+Validate that the code under test doesn't throw an error by comparing
+the error to `Never`:
+
+```swift
+@Test func canAddAttributeToObjectInPositionZero() throws {
+    var array = AttributedArray(list: [1,2,3])
+    #expect(throws: Never.self) {
+        try array.add(attribute: "my attribute", toObjectsIn: 0..<1)
+    }
+    #expect(try array.attributes(forItemAt: 0) == ["my attribute"])
+}
+```
+
+If the closure throws an error, the testing library records an issue.
+If you need the test to stop if the code throws an error, include the
+code inline in the test function instead of wrapping it in an
+`#expect(throws:)` block.
+
+> Note: `#require(throws:Never.self)` is deprecated, because calling
```

Comment:
We don't need to call this out here.

Also, is this style correct for DocC? Will this render as a note callout?

Reply:
Oh neat. We otherwise use:

```swift
- Note: lorem ipsum…
```

Reply:
Removed.